src/extra_config.json

```diff
@@ -0,0 +1,18 @@
+{
+    "katib-ui": {
+        "menuLink": {
+            "type": "item",
+            "link": "/katib/",
+            "text": "Experiments (AutoML)",
+            "icon": "kubeflow:katib"
+        }
+    },
+    "tensorboards-web-app": {
+        "menuLink": {
+            "type": "item",
+            "link": "/tensorboards/",
+            "text": "Tensorboards",
+            "icon": "assessment"
+        }
+    }
+}
```

Comment:
We can make this more dynamic and remove the need for this `extra_config.json` entirely.  Each menuLink item needs `type, link, text, icon`.  This can be passed through the `sidepanel` relation.  So the katib-ui application could relate to the `sidepanel` relation with data: 
```yaml
{
  type: "item",
  link: "/katib/",
  text: "Experiments (AutoML)",
  icon: "kubeflow:katib",  <-- I dont know how this works - maybe this takes extra effort
}
```

This doesn't really add much over what you have here, but it also means some unknown application can also use the relation:

```yaml
{
  type: "item",
  link: "/wherever-i-exist/",
  text: "Some custom application someone else made",
  icon: "my-fancy-icon",
}
```

This way whenever anyone wants to add a menu item, it is driven dynamically through the relation and requires no additional code change here.  

In general, we can make Dashboard into a provider of a sidebar and ignorant to the actual content.